python/ray/WebUI.ipynb

```diff
@@ -80,6 +129,278 @@
     "\n",
     "task_search.on_submit(handle_submit)"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Task Timeline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "path_input = widgets.Button(description=\"View task timeline\")\n",
+    "display(path_input)\n",
+    "\n",
+    "def find_trace2html():\n",
+    "    trace2html = os.path.join(os.path.expanduser(\"~\"), \"catapult\", \"tracing\", \"bin\", \"trace2html\")\n",
+    "    assert os.path.exists(trace2html), \"Could not find catapult, please clone it into your home directory from https://github.com/catapult-project/catapult/tree/master/tracing\"\n",
+    "    return trace2html\n",
+    "\n",
+    "def handle_submit(sender):\n",
+    "    tmp = tempfile.mktemp() + \".json\"\n",
+    "    tmp2 = tempfile.mktemp() + \".html\"\n",
+    "\n",
+    "    print(\"Dumping task profiling data to \" + tmp)\n",
+    "    ray.global_state.dump_catapult_trace(tmp)\n",
+    "    print(\"Converting chrome trace to \" + tmp2)\n",
+    "    trace2html = find_trace2html()\n",
+    "    subprocess.check_output([\"python2\", trace2html, tmp, '--output', tmp2])\n",
+    "    print(\"Opening html file in browser...\")\n",
+    "    subprocess.Popen([\"open\", \"-a\", \"Google Chrome\", tmp2])\n",
+    "\n",
+    "path_input.on_click(handle_submit)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Task Time Series."
```

Comment:
If I start up Ray, don't run any tasks, go to the UI, evaluate the first box, and then click the "Click to run all" button, the "Task Time Series" box fails with

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-7-2598ae40be55> in <module>()
     38 
     39 buckets, earliest, latest, gran = time_series()
---> 40 desired_range = (0, max(buckets)+10)
     41 distr = []
     42 for x in range(len(buckets)):

ValueError: max() arg is an empty sequence
```